src/JitInterface/src/CorInfoImpl.cs

```diff
@@ -183,7 +183,10 @@ private ObjectNode.ObjectData EncodeEHInfo()
                 }
 
                 builder.EmitCompressedUInt((uint)clause.TryOffset);
-                builder.EmitCompressedUInt((uint)(((int)clause.TryLength << 2) | (int)clauseKind));
+
+                // clause.TryLength returned by the JIT is actually end offset...
+                int tryLength = (int)clause.TryLength - (int)clause.TryOffset;
```

Comment:
`CORINFO_EH_CLAUSE.TryLength` means try length in one context, and try end offset in other context.

We paper over it CoreCLR without a blink of an eye: https://github.com/dotnet/coreclr/blob/master/src/vm/jitinterface.cpp#L11328

This is a bug in JIT-EE interface definition. I have opened dotnet/CoreCLR/#3585 on it.